logicalplan/distribute_avg.go

```diff
@@ -57,5 +57,5 @@ func isDistributiveOrAverage(expr *Node, skipBinaryPushdown bool) bool {
 	if aggr, ok := (*expr).(*Aggregation); ok {
 		isAvg = aggr.Op == parser.AVG
 	}
-	return isDistributive(expr, skipBinaryPushdown) || isAvg
+	return isDistributive(expr, skipBinaryPushdown, map[string]struct{}{}) || isAvg
```

Comment:
dont we also need to check external labels here?

Reply:
I have added a test for the logical plan to illustrate what happens with `avg`. I don't think it is necessary to check labels here because this optimizer just rewrites `avg` as `sum / count`. But please let me know if I am missing something.

Reply:
Mh worst case we rewrite it to that in something that doesn't distribute because of label replace. It won't be wrong, we just won't use avg locally then right?

Reply:
Yes, that's right. We can see if it leads to issues and refine it. At this point it should be related to performance and not correctnes.